### DISCUSSION TITLE
Create apps using API and export file

### DISCUSSION THREAD

**Discussion by snvgglebear** [Budibase/budibase#6137] [Help]
Is it possible to create a new app through the api using an app export file? I tried following the instructions in the docs and added "name: [app_name]" at the beginning of the export and got a bad request error.

**Comment by melohagan** [comment]
Hi @gondilon 

The API only takes two body params:

![Screenshot 2022-05-31 at 16 55 10](https://user-images.githubusercontent.com/101575380/171217026-f0a447ab-9437-4188-a899-c3cf96e58e20.png)

You cannot create an app with an app export through the API.

**Reply by snvgglebear** [reply]
Looking at the code, in the application validator function (Which is called by the API as far as I can tell) there is an option for a template string. Is there a way I could save a custom template and then use that parameter? 
exports.applicationValidator = () => {
  // prettier-ignore
  return joiValidator.body(Joi.object({
    _id: OPTIONAL_STRING,
    _rev: OPTIONAL_STRING,
    name: Joi.string().required(),
    url: OPTIONAL_STRING,
    template: Joi.object({
      templateString: OPTIONAL_STRING,
    }).unknown(true),
  }).unknown(true))
}
